bolt-taproot-gossip.md

```diff
@@ -0,0 +1,1083 @@
+# Extension BOLT XX: Taproot Gossip
+
+# Table of Contents
+
+  * [Aim](#aim)
+  * [Overview](#overview)
+  * [Terminology](#terminology)
+  * [Type Definitions](#type-definitions)
+  * [TLV Based Messages](#tlv-based-messages)
+  * [Taproot Channel Proof and Verification](#taproot-channel-proof-and-verification)
+  * [Timestamp fields](#timestamp-fields)
+  * [Bootstrapping Taproot Gossip](#bootstrapping-taproot-gossip)
+  * [Specification](#specification)
+    * [Feature Bits](#feature-bits)
+    * [The `announcement_signatures_2` message](#the-announcement_signatures_2-message)
+    * [The `channel_announcement_2` message](#the-channel_announcement_2-message)
+    * [The `node_announcement_2` message](#the-node_announcement_2-message)
+    * [The `channel_update_2` message](#the-channel_update_2-message)
+  * [Appendix A: Algorithms](#appendix-a-algorithms)
+      * [Partial Signature Calculation](#partial-signature-calculation)
+      * [Partial Signature Verification](#partial-signature-verification)
+  * [Appendix B: Test Vectors](#appendix-b-test-vectors)
+  * [Acknowledgements](#acknowledgements)
+
+## Aim 
+
+This document aims to update the gossip protocol defined in [BOLT 7][bolt-7] to
+allow for advertisement and verification of taproot channels. An entirely new
+set of gossip messages are defined that use [BIP-340][bip-340] signatures and
+that use a purely TLV based schema.
+
+## Overview
+
+The initial version of the Lightning Network gossip protocol as defined in
+[BOLT 7][bolt-7] was designed around P2WSH funding transactions. For these
+channels, the `channel_announcement` message is used to advertise the channel to
+the rest of the network. Nodes in the network use the content of this message to
+prove that the channel is sufficiently bound to the Lightning Network context
+and that it is owned by the nodes advertising the channel. This proof and
+verification protocol is, however, not compatible with SegWit V1 (P2TR) outputs
+and so cannot be used to advertise the channels defined in the
+[Simple Taproot Channel][simple-taproot-chans] proposal. This document thus aims
+to define an updated gossip protocol that will allow nodes to both advertise and
+verify taproot channels. This part of the update affects the
+`announcement_signatures` and `channel_announcement` messages.
+
+The opportunity is also taken to rework the `node_announcement` and
```

Comment:
[From Joost](https://github.com/ellemouton/bolts/pull/2#discussion_r1135117400):

Could you also not take this opportunity and keep using ecdsa signatures for those? That seems to reduce the delta of this proposal significantly, and maybe without loss of functionality for the user?

Reply:
yes we could definitely do this. but: 
- getting to use schnorr sigs for them is quite beneficial for things like multi-key backed nodes
- see the section on "Bootstrapping Taproot Gossip", it is not necessary for nodes to use the new announcement message straight away. They can use the old one for as long as they like.